docs/distribution/nightly.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: developer-doc
+title: Nightly Builds
+category: distribution
+tags: [distribution, release, nightly]
+order: 10
+---
+
+# Nightly Builds
+
+This document describes the infrastructure for automated nightly builds.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Triggering the Build](#triggering-the-build)
+- [Nightly Build Versions](#nightly-build-versions)
+- [Release Notes](#release-notes)
+
+<!-- /MarkdownTOC -->
+
+## Triggering the Build
+
+The build may be triggered on two possible events:
+
+- automatically, at 4am UTC after each working day (that is, on Tuesday to
+  Saturday),
+- manually, if a commit with message containing `[release: nightly]` is pushed
+  to the `main` branch.
+
+The nightly build is based off of the state of the `main` branch at the moment
+when it was triggered.
+
+However, when a nightly build is triggered (by any of the two above conditions),
+it will only proceed if there are any changes. That is, if the current commit is
+the same as the one used for the previous nightly build, the build will not
+proceed, because there are no changes.
+
+Thanks to
+[GitHub's concurrency settings](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency),
+only one nightly or regular release job may be running at a given time. If a
+build is triggered when another one is in progress, it will be pending until the
+first one finishes.
+
+## Nightly Build Versions
+
+The nightly build will modify the version (which is set in `build.sbt`) to
+indicate that this is a specific nightly build.
+
+If the version does not include a `SNAPSHOT` suffix, it is added. Then the
+current date is appended. Moreover, if the build is not the first nightly build
+to be done on a given day, an increasing numeric suffix is appended, to
+differentiate between builds from a single day.
+
+For example, if `build.sbt` specifies the version to be `1.2.3` or
+`1.2.3-SNAPSHOT`, the nightly build based off of that version, triggered on 1st
+of February 2021 will result in version `1.2.3-SNAPSHOT.2021-02-01`. If a
+subsequent build based off of the same version is triggered on the same day it
+will be `1.2.3-SNAPSHOT.2021-02-01.1` etc.
+
+Only the 3 most recent nightly builds are kept in the repository, any older
+builds are removed from releases.
```

Comment:
```suggestion
builds are removed from the releases page.
```

Does this handle removing the corresponding tags, or can we have releases without tags?

Reply:
AFAIK every release has a tag.

We do remove the tags but that is easily changed in the settings of the action removing old releases.

I assume that we want to be removing the tags + that fact should be stated in the doc, correct?

Reply:
We definitely do want to be removing the tags, and it should be stated in the doc!